### PR TITLE
Fix #1299: Move UI module into its own maven module

### LINKED ISSUE #1299: Move UI module into its own maven module
Currently UI module is under catalog-rest-service module, lets move it under its own module

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by akash-jain-10** [comment]
This change will also cause the github actions for [py-generate](https://github.com/open-metadata/OpenMetadata/blob/issue-1299/.github/workflows/py-generate.yml) to fail. Let's update the path to match the changes done here.

**Comment by harshach** [comment]
@akash-jain-10 we can merge it and update the actions. Doesn't need to be in the same

File: .gitignore [hunk 1]
```diff
@@ -67,6 +67,8 @@ secrets
 # ts build info and report
 catalog-rest-service/src/main/resources/ui/webpack
```
Comment (shahsank3t) [inline .gitignore:68]: @darth-coder00 You can remove these path and keep it clean

**Comment by akash-jain-10** [comment]
> @akash-jain-10 we can merge it and update the actions. Doesn't need to be in the same why we need to change make recipe. Its the same path as before

We don't require that change.

File: .gitignore (hunk 1 above)
Comment (darth-coder00) [reply .gitignore:68]: Yes

**Review by darth-coder00 — APPROVED** [review]
LGTM!

**Review by shahsank3t — APPROVED** [review]

**Pull request merged**